internal/commands/launch.go

```diff
@@ -117,17 +119,32 @@ func (l *LaunchCommand) Execute() error {
 		logs.Debugf("Args: %v", runnerCfg.Args)
 		logs.Debugf("Env vars: %v", env.Keys(runnerEnv))
 
-		cmd := exec.Command(runnerCfg.Command, runnerCfg.Args...)
+		ctx, cancelHealthMonitor := context.WithCancel(context.Background())
+		var wg sync.WaitGroup
+
+		cmd := exec.CommandContext(ctx, runnerCfg.Command, runnerCfg.Args...)
 		cmd.Env = runnerEnv
 		cmd.Stdout = os.Stdout
 		cmd.Stderr = os.Stderr
 
-		err = cmd.Run()
-		if err != nil {
-			logs.Infof("Runner process failed: %v", err)
+		if err := cmd.Start(); err != nil {
+			cancelHealthMonitor()
```

Comment:
Do we need this call here? We haven't started the monitoring yet

Reply:
So I actually added this following the linter:

```
make lint
golangci-lint run
internal/commands/launch.go:122:3: lostcancel: the cancelHealthMonitor function is not used on all paths (possible context leak) (govet)
                ctx, cancelHealthMonitor := context.WithCancel(context.Background())
                ^
internal/commands/launch.go:131:4: lostcancel: this return statement may be reached without using the cancelHealthMonitor var defined on line 122 (govet)
                        return fmt.Errorf("failed to start runner process: %w", err)
                        ^
make: *** [lint] Error 1
```

I _think_ this means that even if we don't use the context in the branch where start failed, Go has made an allocation for the context and so the linter is expecting us to clean it up. It's not able to follow control flow to see that we'll exit anyway.